wazimap_ng/datasets/tasks/process_uploaded_file.py

```diff
@@ -13,6 +14,44 @@
 
 logger = logging.getLogger(__name__)
 
+def detect_encoding(filename):
+    detector = UniversalDetector()
+    with open(filename, "rb") as fp:
+        for line in fp:
+            detector.feed(line)
+            if detector.done: break
+    detector.close()
+    return detector.result
+
+def process_file_data(df, dataset, row_number):
+    df = df.applymap(lambda s:s.strip() if type(s) == str else s)
+    datasource = (dict(d[1]) for d in df.iterrows())
+    return loaddata(dataset, datasource, row_number)
+
+def process_csv(dataset, filename, chunksize=1000000):
+    encoding_guess = detect_encoding(filename)
```

Comment:
why are you returning the detector result instead of the encoding?
now this method needs to worry about the format of the detector as well.
if you just returned `detector.result["encoding"]` then `detect_econding` will always return an encoding.

Reply:
you're right - coded it too quickly - will fix